community/detectors/flyte_exposed_console/src/main/java/com/google/tsunami/plugins/rce/ExposedFlyteConsoleDetector.java

```diff
@@ -0,0 +1,198 @@
+@ForWebService
+public final class ExposedFlyteConsoleDetector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  @VisibleForTesting
+  static final String VULNERABILITY_REPORT_PUBLISHER = "TSUNAMI_COMMUNITY";
+  @VisibleForTesting
+  static final String VULNERABILITY_REPORT_ID = "FLYTE_CONSOLE_EXPOSED";
+
+  @VisibleForTesting
+  static final String VULNERABILITY_REPORT_TITLE = "Exposed Flyte Console";
+
+  @VisibleForTesting
+  static final String VULN_DESCRIPTION = "An exposed Flyte Console can lead to severe security risks, "
+      + "including unauthorized access and potential remote code execution (RCE). "
+      + "Ensure that access controls and security measures are properly configured "
+      + "to prevent exploitation. Please refer to the remediation guidance section "
+      + "below for mitigation strategies.";
+
+  @VisibleForTesting
+  static final String RECOMMENDATION = "Please disable public access to your flyte console instance.";
+
+  @VisibleForTesting
+  FlyteProtoClient flyteClient = new FlyteProtoClient();
+
+  private static final int MAX_TIMEOUT_FOR_RCE_IN_SECS = 180;
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+  private final PayloadGenerator payloadGenerator;
+
+  @Inject
+  ExposedFlyteConsoleDetector(
+      @UtcClock Clock utcClock, HttpClient httpClient, PayloadGenerator payloadGenerator) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient).modify().setFollowRedirects(true).build();
+    this.payloadGenerator = checkNotNull(payloadGenerator);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+
+    Builder detectionReport = DetectionReportList.newBuilder();
+    matchedServices.stream()
+        .filter(NetworkServiceUtils::isWebService)
+        .filter(this::isFlyteConsole)
+        .forEach(
+            networkService -> {
+              if (isVulnerable(networkService)) {
+                detectionReport.addDetectionReports(
+                    buildDetectionReport(
+                        targetInfo,
+                        networkService,
+                        "Flyte Console is misconfigured and can be accessed publicly, potentially leading to Remote Code Execution (RCE)."
+                            + " Tsunami security scanner confirmed this by sending an HTTP request"
+                            + " with a test connection API and receiving the corresponding callback"
+                            + " on the tsunami callback server.",
+                        Severity.CRITICAL));
+              }
+            });
+    return detectionReport.build();
+  }
+
+  public boolean isFlyteConsole(NetworkService networkService) {
```

Comment:
The fingerprint phase is incomplete. I'm noticing that there is a specific fingerprinter for this service. You can implement it in order to have a reliable fingerprint phase.

Reply:
Added the fingerprint logic.